backend/main.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/uuid"
+)
+
+// Directory to save uploaded files
+const uploadDir = "uploads"
+
+func uploadHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		http.Error(w, "Only POST method is allowed", http.StatusMethodNotAllowed)
+		return
+	}
+
+	// Set file size limit to 100MB
+	err := r.ParseMultipartForm(100 << 20)
```

Comment:
This method of enforcing 100 MB limit is very memory inefficient because -
1. it till load the whole request (which can be >100MB) in golang memory and then parse it.
    if 10 such request come in at the same time, we will waste 1 GB only in checking for file limit.
    we should go for 'streaming' wherever possible.
    Try with this approach
    references:
    [MaxBytesReader](https://pkg.go.dev/net/http#MaxBytesReader)
    [MaxBytesError](https://pkg.go.dev/net/http#MaxBytesError)
    
    ```
    	// Limit the size of the HTTP request body
	r.Body = http.MaxBytesReader(w, r.Body, maxUploadSize)

	// Open destination file for writing
	filePath := "uploaded_file"
	dst, err := os.Create(filePath)
	if err != nil {
		http.Error(w, "Unable to create file", http.StatusInternalServerError)
		return
	}
	defer func() {
		dst.Close()
	}()

	// Stream file content to disk
	_, err = io.Copy(dst, r.Body)
	if err != nil {
		// Check for MaxBytesError
		var maxBytesErr *http.MaxBytesError
		if errors.As(err, &maxBytesErr) {
			http.Error(w, "File size exceeds the 100 MB limit", http.StatusRequestEntityTooLarge)
		} else {
			http.Error(w, "Failed to save file: "+err.Error(), http.StatusInternalServerError)
		}

		// Delete the partially written file
		os.Remove(filePath)
		return
	}
```
 2.  use this approach  r.Body = http.MaxBytesReader(w, r.Body, 100<<20)